operator/kafkacontroller/create.go

```diff
@@ -0,0 +1,67 @@
+package kafkacontroller
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	exoscalev1 "github.com/vshn/provider-exoscale/apis/exoscale/v1"
+	"github.com/vshn/provider-exoscale/operator/mapper"
+
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	exoscaleapi "github.com/exoscale/egoscale/v2/api"
+	"github.com/exoscale/egoscale/v2/oapi"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+)
+
+// Create idempotently creates a Kafka instance.
+// It will not return an "already exits" error.
+func (c connection) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	log := controllerruntime.LoggerFrom(ctx)
+	log.V(1).Info("creating resource")
+
+	instance, ok := mg.(*exoscalev1.Kafka)
+	if !ok {
+		return managed.ExternalCreation{}, fmt.Errorf("invalid managed resource type %T for kafka connection", mg)
```

Comment:
I might be overly cautious here. I generally want to avoid panics in controllers.

Yes, this would be a bug in crossplane and should never happen, but if it ever does I'm not sure if it's smart to panic as that would cause all other controllers to stop working too.